src/utils/daos_hdlr.c

```diff
@@ -2948,6 +2963,465 @@ fs_copy_hdlr(struct cmd_args_s *ap)
+			}
+			d_iov_set(&diov, (void *)dkey, dkey_kds[j].kd_key_len);
+
+			/* enumerate and parse akeys */
+			rc = cont_copy_list_akeys(src_oh, dst_oh, diov);
+			ptr += dkey_kds[j].kd_key_len;
+		}
+	}
+out:
+	return rc;
+}
+
+static int
+cont_copy_parse(struct duns_attr_t *dattr,
+		uuid_t *pool,
+		uuid_t *cont,
+		char *arg)
+{
+	int	rc = 0;
+	char	*saveptr = NULL;
+	char	*pool_str = NULL;
+	char	*cont_str = NULL;
+
+	/* check for DAOS pool/cont or UNS path */
+	rc = duns_resolve_path(arg, dattr);
+	if (rc != 0) {
+		pool_str = strtok_r(arg, "/", &saveptr);
+		rc = uuid_parse(pool_str, *pool);
+		if (rc == 0) {
+			cont_str = strtok_r(NULL, "/", &saveptr);
+			rc = uuid_parse(cont_str, *cont);
+			if (rc != 0) {
+				fprintf(stderr, "failed to parse cont "
+					"uuid: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+		} else {
+			fprintf(stderr, "failed to parse uuid or "
+				"path: %d\n", rc);
+			D_GOTO(out, rc = EINVAL);
+		}	
+	} else {
+		uuid_copy(*pool, dattr->da_puuid);
+		uuid_copy(*cont, dattr->da_cuuid);
+	}
+out:
+	return rc;
+}
+
+int
+cont_copy_hdlr(struct cmd_args_s *ap)
+{
+	int			rc = 0;
+	int			i = 0;
+	daos_cont_info_t	src_cont_info;
+	daos_cont_info_t	dst_cont_info;
+	static const int	OID_ARR_SIZE = 50;
+ 	daos_obj_id_t		oids[OID_ARR_SIZE];
+ 	daos_anchor_t		anchor;
+	uint32_t		oids_nr;
+ 	daos_handle_t		toh = NULL;
+	daos_epoch_t		epoch;
+	struct			duns_attr_t dst_dattr = {0};
+	struct			duns_attr_t src_dattr = {0};
+	struct			copy_args cp_args = {0};
+	bool			is_posix_copy = false;
+	daos_handle_t		oh = DAOS_HDL_INVAL;
+	daos_handle_t		dst_oh = DAOS_HDL_INVAL;
+	struct file_dfs		src_cp_type = {0};
+	struct file_dfs		dst_cp_type = {0};
+	daos_epoch_range_t	epr;
+
+	file_set_defaults_dfs(&src_cp_type);
+	file_set_defaults_dfs(&dst_cp_type);
+	rc = cont_copy_parse(&src_dattr, &cp_args.src_p_uuid,
+			     &cp_args.src_c_uuid, ap->src);
+	if (rc != 0) {
+		fprintf(stderr, "failed to parse source argument: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+	rc = cont_copy_parse(&dst_dattr, &cp_args.dst_p_uuid,
+			     &cp_args.dst_c_uuid, ap->dst); 
+	if (rc != 0) {
+		fprintf(stderr, "failed to parse destination "
+			"argument: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+	rc = copy_connect(is_posix_copy, &dst_cp_type, &src_cp_type,
+			  &cp_args, ap->sysname, &src_cont_info,
+			  &dst_cont_info);
+	if (rc != 0) {
+		D_GOTO(out_disconnect, rc);
+	}
+
+	/* use snapshot if one was passed in, otherwise create one */
+	if (ap->epc != 0) {
+		epoch = ap->epc;
+	} else {
+		rc = daos_cont_create_snap_opt(cp_args.src_coh, &epoch,  NULL,
+				DAOS_SNAP_OPT_CR | DAOS_SNAP_OPT_OIT,
+			       	NULL);
+		if (rc) {
+			fprintf(stderr, "failed to create snapshot: %d\n", rc);
+			D_GOTO(out_disconnect, rc);
+		}
+	}
+	rc = daos_oit_open(cp_args.src_coh, epoch, &toh, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "failed to open object iterator\n");
+		D_GOTO(out_disconnect, rc);
+	}
+	memset(&anchor, 0, sizeof(anchor));
+	while (1) {
+ 		oids_nr = OID_ARR_SIZE;
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs